nvflare/app_opt/xgboost/histogram_based/executor.py

```diff
@@ -264,6 +263,9 @@ def train(self, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal) -
 
         try:
             with xgb.collective.CommunicatorContext(**communicator_env):
+                if not self.train_data or not self.val_data:
+                    self.train_data, self.val_data = self.data_loader.load_data(self.client_id, self.app_dir)
+
```

Comment:
Does load_data need to be called inside "with xgb.collective.CommunicatorContext(**communicator_env):"?

If yes, then this is good.
If not, we should move it out

Reply:
Yep, the creation of the Dmatrix in the dataloader needs to occur inside the CommunicatorContext

Reply:
Sorry, I don't understand it fully.
Is it because the "split_mode" argument only works within CommunicatorContext?
(`xgb.DMatrix(valid_path + f"?format=csv{label}", data_split_mode=1)`)